TrackingProduction/Fun4All_raw_hit_KFP.C

```diff
@@ -0,0 +1,713 @@
+#include <trackingqa/TpcClusterQA.h>
+#include <tpcqa/TpcRawHitQA.h>
+#include <trackingqa/TpcSeedsQA.h>
+
+#include <trackingdiagnostics/TrackResiduals.h>
+#include <trackingdiagnostics/TrkrNtuplizer.h>
+#include <trackingdiagnostics/KshortReconstruction.h>
+#include <kfparticle_sphenix/KFParticle_sPHENIX.h>
+
+#include <stdio.h>
+
+R__LOAD_LIBRARY(libkfparticle_sphenix.so)
+R__LOAD_LIBRARY(libfun4all.so)
+R__LOAD_LIBRARY(libffamodules.so)
+R__LOAD_LIBRARY(libphool.so)
+R__LOAD_LIBRARY(libcdbobjects.so)
+R__LOAD_LIBRARY(libmvtx.so)
+R__LOAD_LIBRARY(libintt.so)
+R__LOAD_LIBRARY(libtpc.so)
+R__LOAD_LIBRARY(libmicromegas.so)
+R__LOAD_LIBRARY(libTrackingDiagnostics.so)
+R__LOAD_LIBRARY(libtrackingqa.so)
+R__LOAD_LIBRARY(libtpcqa.so)
+
+namespace HeavyFlavorReco
+{
+  int VERBOSITY = 0;
+
+  std::string output_dir = "./"; //Top dir of where the output nTuples will be written
+  std::string kfp_header = "outputKFParticle_";
+  std::string processing_folder = "inReconstruction/";
+  std::string trailer = ".root";
+
+  std::string pipi_decay_descriptor = "K_S0 -> pi^+ pi^-"; //See twiki on how to set this
+  std::string pipi_reconstruction_name = "pipi_reco"; //Used for naming output folder, file and node
+  std::string pipi_output_reco_file;
+  std::string pipi_output_dir;
+
+  std::string ppi_decay_descriptor = "[Lambda0 -> proton^+ pi^-]cc"; //See twiki on how to set this
+  std::string ppi_reconstruction_name = "ppi_reco"; //Used for naming output folder, file and node
+  std::string ppi_output_reco_file;
+  std::string ppi_output_dir;
+
+    bool save_tracks_to_DST = false;
+  bool dont_use_global_vertex = true;
+  bool require_track_and_vertex_match = true;
+  bool save_all_vtx_info = true;
+  bool constrain_phi_mass = false;
+  bool constrain_lambda_mass = false;
+  bool constrain_D_mass = false;
+  bool use_2D_matching = false;
+  bool get_trigger_info = true;
+  bool get_detector_info = true;
+  bool get_dEdx_info = true;
+  bool constrain_to_primary_vertex = true;
+  bool use_pid = true;
+  float pid_frac = 0.4;
+}; // namespace HeavyFlavorReco
+
+using namespace HeavyFlavorReco;
+
+void create_hf_directories(std::string reconstruction_name, std::string &final_output_dir, std::string &output_reco_file)
+{
+  std::string output_file_name = kfp_header + reconstruction_name + trailer;
+  final_output_dir = output_dir + reconstruction_name + "/";
+  std::string output_reco_dir = final_output_dir + processing_folder;
+  output_reco_file = output_reco_dir + output_file_name;
+
+  std::string makeDirectory = "mkdir -p " + output_reco_dir;
+  system(makeDirectory.c_str());
+}
+
+
+void end_kfparticle(std::string full_file_name, std::string final_path)
+{
+  ifstream file(full_file_name.c_str());
+  if (file.good())
+  {
+    string moveOutput = "mv " + full_file_name + " " + final_path;
+    system(moveOutput.c_str());
+  }
+}
+
+void Fun4All_raw_hit_KFP(
+    const int nEvents = 100,
+    const int runnumber = 53871,
```

Comment:
The run number and segment are extracted from the filenames in line 271, we don't need to pass these as arguments